src/dscanner/analysis/line_length.d

```diff
@@ -6,176 +6,143 @@
 module dscanner.analysis.line_length;
 
 import dscanner.analysis.base;
-
-import dparse.ast;
-import dparse.lexer;
-
-import std.typecons : tuple, Tuple;
+import dmd.tokens : Token, TOK;
 
 /**
  * Checks for lines longer than `max_line_length` characters
  */
-final class LineLengthCheck : BaseAnalyzer
+extern (C++) class LineLengthCheck : BaseAnalyzerDmd
 {
 	mixin AnalyzerInfo!"long_line_check";
+	private enum KEY = "dscanner.style.long_line";
+	immutable string msg;
+
+	private Token[] tokens;
+	private immutable int maxLineLength;
+	private uint currentLine = 1;
+	private int currentLineLen;
 
-	///
-	this(BaseAnalyzerArguments args, int maxLineLength)
+	extern (D) this(string fileName, bool skipTests = false, int maxLineLength = 120)
```

Comment:
Check this out: gdc won't compile this code without setting a default value for the `maxLineLength` var